### PR TITLE
[AERIE-1893 ] Remove deprecated sim results response code

### PR DESCRIPTION
**Pull request by skovati**
* **Tickets addressed:** AERIE-1893
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

Thanks to work on previous PRs (#209, #211, #189), simulation results are now entirely separated from the `simulate` Hasura action. Constraint violations are now queried with `constraintViolations`, resource samples with `resourceSamples`, and simulated activities directly from the database. UI has been [updated](https://github.com/NASA-AMMOS/aerie-ui/pull/38) to use new queries.

## Verification

Manually tested UI to ensure functionality is maintained.

```graphql
query {
  simulate(planId: 1) {
    status
    reason
    results
  }
}
```
used to get a response of
```json
{
  "data": {
    "simulate": {
      "status": "complete",
      "results": {
        "unfinishedActivities": {},
        "start": "2022-023T12:12:12.000000",
        "activities": {
           ...
        "constraints": {
          "model/no waste": [
            ...
        "events": [
          {
            "graph": {
              "suffix": {
                ...
```

now
```graphql
query {
  simulate(planId: 1) {
    reason
    status
  }
}
```

receives
```json
{
  "data": {
    "simulate": {
      "status": "complete"
    }
  }
}
```

## Documentation

Need to rework existing documentation on `simulate` responses, and point towards new API calls.

## Future work

Opens the door for simulation streaming to database, as clients are no longer dependent on monolithic `simulate` call

### CODE REVIEW

**Comment by mattdailis** [comment]
As you mention in your documentation section, I think we'll need to do a fair bit of legwork to come up with exactly what GraphQL queries can be used to retrieve all of the information that used to be part of simulation results. I'm thinking specifically about the activity start times, durations, input and output attributes, finished/unfinished state, as well as topics/events. Once we come up with these queries, we can help @camargo update the UI.

**Comment by camargo** [comment]
@mattdailis The UI is already off this query :)

**Comment by skovati** [comment]
Yep! UI is using our new queries, so the `results` data we're returning to the `simulate` query with is entirely unused; this PR just removes that wasted compute.

**Review by mattdailis — APPROVED** [review]
Fantastic 🎉 PRs that mostly remove code feel so good.

**Comment by dyst5422** [comment]
Are we at least returning the simulation_dataset id?

**Comment by camargo** [comment]
@dyst5422 I agree that would be useful. Clients could use the `simulation_dataset` id returned from the simulate action to immediately subscribe to the status of the simulation results they are waiting on. Can we make a ticket for this and handle it in a separate PR?

**Review by camargo — APPROVED** [review]

**Comment by mattdailis** [comment]
I'll say hold off on merging until we know the story from the command expansion side - those tests don't run automatically with PRs, so we should run them manually and see if they hit any snags.

**Comment by dyst5422** [comment]
Will go in to rerun command expansion tests and ensure we are updated on tests there.

**Comment by dyst5422** [comment]
~~Getting errors on each of the simulations~~
```
Error: java.lang.NoSuchMethodError: 'void gov.nasa.jpl.aerie.contrib.models.Accumulator.subtract(double)'
```

~~Full trace from worker~~
```
java.lang.NoSuchMethodError: 'void gov.nasa.jpl.aerie.contrib.models.Accumulator.subtract(double)'
2022-06-16T17:33:31.929963004Z 	at gov.nasa.jpl.aerie.banananation.activities.PeelBananaActivity.run(PeelBananaActivity.java:36)
2022-06-16T17:33:31.929965703Z 	at gov.nasa.jpl.aerie.banananation.generated.activities.PeelBananaActivityMapper.lambda$createTask$3(PeelBananaActivityMapper.java:124)
2022-06-16T17:33:31.929968135Z 	at gov.nasa.jpl.aerie.merlin.framework.ThreadedTask$ThreadedTaskHandle.run(ThreadedTask.java:144)
2022-06-16T17:33:31.929970428Z 	at gov.nasa.jpl.aerie.merlin.framework.ThreadedTask.lambda$beginAsync$0(ThreadedTask.java:99)
2022-06-16T17:33:31.929972721Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-06-16T17:33:31.929974940Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-06-16T17:33:31.929977189Z 	at java.base/java.lang.Thread.run(Thread.java:833)
2022-06-16T17:33:31.930040242Z 	Suppressed: gov.nasa.jpl.aerie.merlin.framework.ThreadedTask$TaskFailureException: Observed task thread failure from driver thread
2022-06-16T17:33:31.930045678Z 		at gov.nasa.jpl.aerie.merlin.framework.ThreadedTask.step(ThreadedTask.java:65)
2022-06-16T17:33:31.930048227Z 		at gov.nasa.jpl.aerie.merlin.driver.SimulationDriver$1.step(SimulationDriver.java:150)
2022-06-16T17:33:31.930050530Z 		at gov.nasa.jpl.aerie.merlin.driver.engine.SimulationEngine.stepEffectModel(SimulationEngine.java:206)
2022-06-16T17:33:31.930053116Z 		at gov.nasa.jpl.aerie.merlin.driver.engine.SimulationEngine.stepTaskHelper(SimulationEngine.java:187)
2022-06-16T17:33:31.930055388Z 		at gov.nasa.jpl.aerie.merlin.driver.engine.SimulationEngine.stepTask(SimulationEngine.java:175)
2022-06-16T17:33:31.930057818Z 		at gov.nasa.jpl.aerie.merlin.driver.engine.SimulationEngine.performJob(SimulationEngine.java:152)
2022-06-16T17:33:31.930060046Z 		at gov.nasa.jpl.aerie.merlin.driver.engine.SimulationEngine.lambda$performJobs$0(SimulationEngine.java:136)
2022-06-16T17:33:31.930152382Z 		at gov.nasa.jpl.aerie.merlin.driver.engine.TaskFrame.run(TaskFrame.java:45)
2022-06-16T17:33:31.930158716Z 		at gov.nasa.jpl.aerie.merlin.driver.engine.SimulationEngine.performJobs(SimulationEngine.java:135)
2022-06-16T17:33:31.930161219Z 		at gov.nasa.jpl.aerie.merlin.driver.SimulationDriver.simulate(SimulationDriver.java:89)
2022-06-16T17:33:31.930172093Z 		at gov.nasa.jpl.aerie.merlin.server.models.MissionModelFacade.simulate(MissionModelFacade.java:37)
2022-06-16T17:33:31.930174642Z 		at gov.nasa.jpl.aerie.merlin.server.services.LocalMissionModelService.runSimulation(LocalMissionModelService.java:208)
2022-06-16T17:33:31.930176837Z 		at gov.nasa.jpl.aerie.merlin.server.services.SynchronousSimulationAgent.simulate(SynchronousSimulationAgent.java:69)
2022-06-16T17:33:31.930178957Z 		at gov.nasa.jpl.aerie.merlin.worker.MerlinWorkerAppDriver.main(MerlinWorkerAppDriver.java:75)
```

~~EDIT: Might be stale build artifacts, checking that now~~

Yes just a stale build issue.

**Comment by mattdailis** [comment]
> Getting errors on each of the simulations
> ...
> EDIT: Might be stale build artifacts, checking that now

That would be my guess as well - perhaps wiping out the banananation build directory would help?

**Comment by dyst5422** [comment]
Tests updated and passing, this has my green light

**Review by dyst5422 — APPROVED** [review]

**Comment by skovati** [comment]
Good catch Matt. Any reason we don't run command expansion tests in our CI?

**Comment by dyst5422** [comment]
> Good catch Matt. Any reason we don't run command expansion tests in our CI?

I haven't gotten around to setting it up yet - its the next thing on my board and I'll be working on it today

**Comment by skovati** [comment]
Rebased onto upstream develop

**Pull request merged**
